bundles/org.eclipse.jface/src/org/eclipse/jface/viewers/AbstractTreeViewer.java

```diff
@@ -102,8 +112,24 @@ public abstract class AbstractTreeViewer extends ColumnViewer {
 	private int expandToLevel = 0;
 
 	/**
-	 * Indicates if filters should be checked to determine expandability of
-	 * a tree node.
+	 * How many levels to autoexpand in a path of consecutive single children. 0
+	 * will disable the feature, while a negative value (<code>ALL_LEVELS</code>)
+	 * will expand to infinity
+	 */
+	private int autoExpandOnSingleChildLevel = 0;
```

Comment:
Should this be called `autoExpandOnSingleChildLevels`?` It does not indicate a level but a number of levels. Same applied to the according method.

Reply:
Since this behaves like `setAutoExpandLevel` with a special condition, I don't see what warrants the inconcistency in plural/singular. "Level" refers to how deep a recursive path with singular descendants will be expanded - what is the meaning behind talking about "multiple levels"?

Reply:
I mentioned this, because the behavior is actually different from `expandToLevel` / `setAutoExpandLevel`. That existing expand level variable defines the level to which auto-expansion is performed. It does not define a number of (relative) levels to expand but defines precisely one level, to whcih expansion is done. The `autoExpandOnSingleChildLevel`, however, does not define a specific level, to which the expansion is performed, but defines _a number of levels_ to expand.

Reply:
Now I understand your comment, I have addressed it in the newest forcepush